File: src/main/java/com/manson/domain/fo76/items/enums/ArmorGrade.java
package com.manson.domain.fo76.items.enums;

import com.fasterxml.jackson.annotation.JsonValue;
import java.util.Arrays;
import org.apache.commons.lang3.StringUtils;

public enum ArmorGrade {
    Light("Light"), Sturdy("Sturdy"), Heavy("Heavy"), Unknown("Unknown");

    @JsonValue
    private final String value;

    ArmorGrade(String value) {
        this.value = value;
    }

    public static ArmorGrade fromString(String input) {
        if (StringUtils.isBlank(input)) {
            return Unknown;
        }
        return Arrays.stream(ArmorGrade.values())
            .filter(x -> StringUtils.equalsIgnoreCase(x.value, input))
            .findFirst()
            .orElse(Unknown);
    }

    public String getValue() {
        return value;
    }
}

File: src/main/kotlin/com/manson/domain/fo76/items/enums/ArmorGrade.kt
package com.manson.domain.fo76.items.enums

import com.fasterxml.jackson.annotation.JsonValue
import org.apache.commons.lang3.StringUtils

enum class ArmorGrade(@get:JsonValue val value: String) {
    Light("Light"), Sturdy("Sturdy"), Heavy("Heavy"), Unknown("Unknown");

    companion object {
        @JvmStatic
        fun fromString(input: String?): ArmorGrade {
            if (StringUtils.isBlank(input)) {
                return Unknown
            }
            for (grade in values()) {
                if (StringUtils.equalsIgnoreCase(grade.value, input)) {
                    return grade
                }
            }
            return Unknown
        }
    }
}